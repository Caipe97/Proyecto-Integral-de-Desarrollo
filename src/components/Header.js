import React from 'react';
import logo from '../images/logo_small.png';
import { useHistory } from "react-router-dom";
import { logout } from '../store/actions';
import { connect } from 'react-redux';

const mapStateToProps = () => {
  return {}
}

const mapDispatchToProps = (dispatch) => {
  return {
    onLogout: () => dispatch(logout())
  }
}

const Header = (props) => {
  let history = useHistory();
  return ( 
  <header className="header">
    <div style={{ width: '100%', position: 'relative' }}>
      <div className="box">
        <img src={logo} alt="imagen tracking" style={{ width: 100 }} />
        {props.userId > 0 ? <button type="button" onClick={() => {props.onLogout(); history.push("/")}}>Log out</button> : null}
      </div>
    </div>
  </header>
  )
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);