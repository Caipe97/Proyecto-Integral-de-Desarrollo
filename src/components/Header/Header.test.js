import React from 'react';
import { shallow } from 'enzyme'; 
import Header from './Header';

describe('Header tests', () => {
    const historyMock = { push: jest.fn() };
    const mockPropsSignIn = {userId: 1, onLogout: jest.fn()};
    let wrapper;
    const mockPropsSignIn2 = {userId: 0, onLogout: jest.fn()};
    let wrapper2;

    beforeEach(() => {
        wrapper = shallow(<Header {...mockPropsSignIn} history={historyMock}/>);
        wrapper2 = shallow(<Header {...mockPropsSignIn2} history={historyMock}/>);
    
    })

    it('expect to render Header component to render with signing in', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('expect to logout and go to login when pressing the button', () => {
        wrapper.find('[name="link"]').at(0).simulate('click');
        expect(wrapper.instance().props.onLogout).toHaveBeenCalledTimes(1);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(1);
        expect(historyMock.push.mock.calls[0]).toEqual(['/']);
    })
    it('expect to render Header component to user not defined', () => {
        expect(wrapper2).toMatchSnapshot();
    })
    it('expect to logout and go to login  not exists button', () => {
        //expert(wrapper2.find('.img')).toEqual(null);
        //expect(wrapper2.instance().props.userId).toEqual(0);
        expect(wrapper2).toMatchSnapshot();
      //  expect(wrapper.instance().props.onLogout).toHaveBeenCalledTimes(1);
       // expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(1);
        //expect(historyMock.push.mock.calls[0]).toEqual(['/']);
    })
})
