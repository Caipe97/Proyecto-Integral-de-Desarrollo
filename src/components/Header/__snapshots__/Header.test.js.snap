// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests expect to logout and go to login  not exists button 1`] = `
<div
  className="header"
  style={
    Object {
      "background": "#0E4749",
      "paddingBottom": "10px",
      "paddingTop": "10px",
      "top": 0,
    }
  }
>
  <nav
    className="navbar sticky-top navbar-expand-lg navbar-dark p-3"
  >
    <div
      className="container"
    >
      <img
        alt="imagen tracking"
        src="logo_small.png"
        style={
          Object {
            "width": 200,
          }
        }
      />
    </div>
  </nav>
</div>
`;

exports[`Header tests expect to render Header component to render with signing in 1`] = `
<div
  className="header"
  style={
    Object {
      "background": "#0E4749",
      "paddingBottom": "10px",
      "paddingTop": "10px",
      "top": 0,
    }
  }
>
  <nav
    className="navbar sticky-top navbar-expand-lg navbar-dark p-3"
  >
    <div
      className="container"
    >
      <img
        alt="imagen tracking"
        src="logo_small.png"
        style={
          Object {
            "width": 200,
          }
        }
      />
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
        style={
          Object {
            "height": "23px",
          }
        }
      >
        <ul
          className="navbar-nav ms-auto"
        >
          <li>
            <Link
              name="link"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                  "textDecoration": "none",
                }
              }
              to="/"
            >
              Cerrar Sesi√≥n
            </Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Header tests expect to render Header component to user not defined 1`] = `
<div
  className="header"
  style={
    Object {
      "background": "#0E4749",
      "paddingBottom": "10px",
      "paddingTop": "10px",
      "top": 0,
    }
  }
>
  <nav
    className="navbar sticky-top navbar-expand-lg navbar-dark p-3"
  >
    <div
      className="container"
    >
      <img
        alt="imagen tracking"
        src="logo_small.png"
        style={
          Object {
            "width": 200,
          }
        }
      />
    </div>
  </nav>
</div>
`;
